### DISCUSSION TITLE
VS Code formatOnSave changes by Darker leads to file conflicts

### DISCUSSION THREAD

**Discussion by kedhammar** [akaihola/darker#525] [Q&A] [answered]
Hi, I've used Darker disguised as Black in VS Code for a while but am running into some issues adapting to the changes mentioned in #516.

I have the Python and Black Formatting extensions installed in VS Code as well as the following user settings .JSON:
```
{
    "black-formatter.path": ["/Users/alfred.kedhammar/miniconda3/envs/alfred/bin/darker"],
    "[python]": {
        "editor.defaultFormatter": "ms-python.black-formatter",
        "editor.formatOnSave": true
    }
}
```
The first time I press save, the changes by Darker are written to disk but do not show up in the IDE. The second time I press save, I get a conflict in which I can only choose to overwrite with the IDE version. Unfortunately this makes Darker unusable inside VS Code.

Any idea how I can fix this?

**Comment by Ashblaze** [comment, marked as answer]
@kedhammar It might be worth making this an issue, so that there is a proper fix from darker. Also, pasting my workaround from the other discussion post.

As a workaround, it is possible to use "[Run on Save](https://marketplace.visualstudio.com/items?itemName=emeraldwalk.RunOnSave)" or "emeraldwalk" extension from the VSCode Marketplace to automatically trigger darker.

**Steps:**
1. Set "Default Formatter" as "None" in your VS Code settings.
2. Uncheck "Format on Save" (OR) remove/disable Black (and all other) formatter extension from VS Code.
3. Install extension "Run on Save" (emeraldwalk) from marketplace.
4. Add darker executable's folder location to system path.
5. Add the following to settings.json (you can modify this to suit your needs):
```
"emeraldwalk.runonsave": {
    "commands": [
        {
            "match": ".py",
            "cmd": "darker \"${file}\" -S"
        }
    ]
}
```

**Reply by kedhammar** [reply]
Thank you!

**Comment by akaihola** [comment]
I'm collecting VSCode related issues in #527.
